### PR TITLE
Resume transfer erc20 to near

### PR DESCRIPTION
**Pull request by ailisp**
Works locally:
```
➜  environment git:(resume-transfer) ✗ node index.js transfer-eth-erc20-to-near --amount 1000 \
--eth-sender-sk 0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501200 \
--near-receiver-account eth2nearprover --near-master-account nearfuntoken \
2>&1 | tee -a /tmp/eth2neartransfer.out
Approving token transfer.
Approved token transfer.
Transferring tokens from the ERC20 account to the token locker account.
Success tranfer to locker
Transferring 1000 tokens from 0x7cC4B1851c35959D34e635A470F6b5C43bA3C9c9 ERC20. From 0xDf08F82De32B8d460adbE8D72043E3a7e25A3B39 sender to eth2nearprover recipient
Eth2NearClient is currently at block 152. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 154. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 154. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
^C
➜  environment git:(resume-transfer) ✗ node index.js transfer-eth-erc20-to-near --amount 1000 \
--eth-sender-sk 0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501200 \
--near-receiver-account eth2nearprover --near-master-account nearfuntoken \
2>&1 | tee -a /tmp/eth2neartransfer.out
Transferring 1000 tokens from 0x7cC4B1851c35959D34e635A470F6b5C43bA3C9c9 ERC20. From 0xDf08F82De32B8d460adbE8D72043E3a7e25A3B39 sender to eth2nearprover recipient
Eth2NearClient is currently at block 156. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 156. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 157. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 158. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 159. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 160. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 161. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 161. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 162. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Eth2NearClient is currently at block 163. Waiting for block 154 to be confirmed. Sleeping for 10 sec.
Balance of eth2nearprover before the transfer is 2000
TxHash 4KVuyz91eUxWCULzZX4VtDZvGLYLXnEpjG5TpKwjG5oQ
[
  'token: 7cc4b1851c35959d34e635a470f6b5c43ba3c9c9; sender: df08f82de32b8d460adbe8d72043e3a7e25a3b39; amount: 1000; recipient: eth2nearprover',
  'Refunding 42800000000000000000000 tokens for storage'
]
Transferred
Balance of eth2nearprover after the transfer is 3000
```

### CODE REVIEW

File: environment/commands/transfer-eth-erc20-to-near.js [hunk 1]
```diff
@@ -96,89 +236,36 @@ class TransferETHERC20ToNear {
...
 
-                break;
-            }
-        }
+        const clientAccount = RainbowConfig.getParam('eth2near-client-account');
+        const ethClientContract = new Eth2NearClientContract(nearMasterAccount, clientAccount);
 
-        if (!logFound) {
-            console.log(`ERROR log not found for event ${lockedEvent}`);
+        if (transferLog.finished === undefined) {
```
Comment (nearmax) [inline environment/commands/transfer-eth-erc20-to-near.js:251]: Nice pattern.

File: environment/commands/transfer-eth-erc20-to-near.js [hunk 2]
```diff
@@ -96,89 +236,36 @@ class TransferETHERC20ToNear {
...
+            await TransferETHERC20ToNear.approve({ ethERC20Contract, amount, ethSenderAccount });
+            transferLog = TransferETHERC20ToNear.loadTransferLog();
+        }
+        if (transferLog.finished === 'approve') {
+            await TransferETHERC20ToNear.lock({ ethTokenLockerContract, amount, nearReceiverAccount, ethSenderAccount });
```
Comment (nearmax) [inline environment/commands/transfer-eth-erc20-to-near.js:256]: Is there a reason we are passing args as an object?

Comment (ailisp) [reply environment/commands/transfer-eth-erc20-to-near.js:256]: Major benefit is no worry about sequence of arguments so it's less error prune when there're many args.

**Comment by ailisp** [comment]
fail is transfer from erc20 to near, irrevelant to this PR and fixed in #227

**Pull request merged**